source/virtualBuffers/__init__.py

```diff
@@ -109,14 +109,12 @@ def obj(self):
 
 	@property
 	def label(self):
-		if self.itemType == "landmark":
-			attrs = self.textInfo._getControlFieldAttribs(self.vbufFieldIdentifier[0], self.vbufFieldIdentifier[1])
-			name = attrs.get("name", "")
-			if name:
-				name += " "
-			return name + aria.landmarkRoles[attrs["landmark"]]
-		else:
-			return super(VirtualBufferQuickNavItem,self).label
+		value = super(VirtualBufferQuickNavItem,self).label
+		if self.itemType is "heading":
+			# Explicitly handle headings here as well, to avoid requesting the Control field attribs

```

Comment:
Another way of doing this so you can rely entirely on `_getLabelForProperties` would be to make a lazy fetcher for control field attributes. This might be overkill, but I'll let you decide. It would go something like this:

```
attrs = {}
def propertyGetter(prop):
	if not attrs:
		# Lazily fetch the attributes the first time they're needed.
		# We do this because we don't want to do this if they're not needed at all.
		attrs.update(self.textInfo._getControlFieldAttribs(self.vbufFieldIdentifier[0], self.vbufFieldIdentifier[1]))
	return attrs.get(prop)
```